pkg/jx/cmd/step_create_task.go

```diff
@@ -403,6 +403,11 @@ func (o *StepCreateTaskOptions) Run() error {
 	if err != nil {
 		return errors.Wrap(err, "failed to generate Tekton CRD")
 	}
+
+	if o.ViewSteps {
```

Comment:
Would it not make more sense to remove

```golang
	if o.ViewSteps {
		return nil, nil, nil, nil, nil, o.viewSteps(tasks...)
	}
```

from within [GenerateTektonCRDs ](https://github.com/jenkins-x/jx/blob/master/pkg/jx/cmd/step_create_task.go#L608) and move it here:

```golang
if o.ViewSteps {
   o.viewSteps(tasks...)
   return nil
}
```

Right now the code is confusing to me, since just returning _nil_ keeps me wondering where the "viewing" part is. One needs to search for all usages of `ViewSteps` to understand.

Reply:
I guess this was created like that just to save some computing only.

Reply:
> I guess this was created like that just to save some computing only.

Probably. Feels a bit of premature optimisation.

Reply:
indeed @hferentschik 
I've just uploaded those changes